systems/plants/RigidBodyManipulator.cpp

```diff
@@ -303,49 +309,15 @@ RigidBodyManipulator::~RigidBodyManipulator(void)
 
 
 // Note:  this method is gross and should be scheduled for deletion upon switching to the new kinsol
-void RigidBodyManipulator::resize(int ndof, int num_featherstone_bodies, int num_rigid_body_objects, int num_rigid_body_frames)
+void RigidBodyManipulator::resize(int ndof, int num_rigid_body_objects, int num_rigid_body_frames)
 {
-  int last_NB = NB, last_num_bodies = num_bodies;
+  int last_num_bodies = num_bodies;
 
   num_positions = ndof;
 
-  if (num_featherstone_bodies<0)
-    NB = ndof;
-  else
-    NB = num_featherstone_bodies;
-
-  pitch.conservativeResize(NB);
-  parent.conservativeResize(NB);
-  dofnum.conservativeResize(NB);
-  damping.conservativeResize(NB);
-  coulomb_friction.conservativeResize(NB);
-  static_friction.conservativeResize(NB);
-  coulomb_window.conservativeResize(NB);
-
-  // note: these are std:vectors (and the above are eigen vectors)
-  Xtree.resize(NB);
-  I.resize(NB);
-  S.resize(NB);
-  Xup.resize(NB);
-  v.resize(NB);
-  avp.resize(NB);
-  fvp.resize(NB);
-  IC.resize(NB);
-  for(int i=last_NB; i < NB; i++) {
-    Xtree[i] = MatrixXd::Zero(6,6);
-    I[i] = MatrixXd::Zero(6,6);
-    S[i] = VectorXd::Zero(6);
-    Xup[i] = MatrixXd::Zero(6,6);
-    v[i] = VectorXd::Zero(6);
-    avp[i] = VectorXd::Zero(6);
-    fvp[i] = VectorXd::Zero(6);
-    IC[i] = MatrixXd::Zero(6,6);
-  }
-
-  if (num_rigid_body_objects<0)
-    num_bodies = NB+1;  // this was my old assumption, so leave it here as the default behavior
-  else
-    num_bodies = num_rigid_body_objects;
+  if (num_rigid_body_objects < 0)
+    throw runtime_error("number of rigid body objects must be positive");
+  num_bodies = num_rigid_body_objects;
 
   I_world.resize(num_bodies);
```

Comment:
possibly for the next pull request:  can we move these into the RigidBody class, and get rid of the resize method completely?